src/ParallelRun.jl

```diff
@@ -296,8 +299,388 @@ function load_images(rcfs, stagedir)
+
+- nprocthreads - number of threads to which to distribute processing the sources.
+- n_sources - the number of total sources to process.
+
+Returns:
+- An array of vectors representing the workload of each thread ([thread][batch][sources]). 
+  Note for partition equally, there is only 1 batch.
+
+"""
+function partition_equally(nprocthreads, n_sources)
+    Log.info("Starting basic source partitioning...")
+    tic()
+    n_sources_per_thread = convert(Int64, floor(n_sources / nprocthreads))
+    thread_sources_assignment = Array(Vector{Vector{Int64}}, nprocthreads)
+    n_sources_assigned = 0
+    for thread = 1:nprocthreads
+        start_source = (thread-1) * n_sources_per_thread
+        end_source = thread * n_sources_per_thread
+        if thread == nprocthreads
+            end_source = n_sources
+        end
+        # Only 1 batch for partitioning equally
+        thread_sources_assignment[thread] = Vector{Vector{Int64}}(1)
+        thread_sources_assignment[thread][1] = Vector{Int64}(end_source-start_source)
+        for source_assignment = start_source:end_source-1
+            thread_sources_assignment[thread][1][source_assignment-start_source+1] = source_assignment+1
+            n_sources_assigned += 1
+        end
+    end
+    @assert n_sources_assigned == n_sources
+    Log.info("Finished basic source partitioning. Elapsed time: $(toq()) seconds")
+    thread_sources_assignment
+end
+
+"""
+Like one_node_infer, uses multiple threads on one node to fit the Celeste 
+model over numerous iterations. 
+TODO max - refactor into different source file (E.G: Cyclades.jl)? Maybe also rename?
+
+- rcfs: Array of run, camcol, field triplets that the source occurs in.
+- box: a bounding box specifying a region of sky
+
+Returns:
+
+- Dictionary of results, keyed by SDSS thing_id.
+"""
+function one_node_infer_multi_iter(rcfs::Vector{RunCamcolField},
+                                   stagedir::String;
+                                   cyclades_partition=true,
+                                   n_iters=10,
+                                   objid="",
+                                   box=BoundingBox(-1000., 1000., -1000., 1000.),
+                                   primary_initialization=true,
+                                   reserve_thread=Ref(false),
+                                   thread_fun=phalse,
+                                   timing=InferTiming())
+    nprocthreads = nthreads()
+    if reserve_thread[]
+        nprocthreads = nprocthreads-1
+    end
+    Log.info("Running with $(nprocthreads) threads")
+    
+    # Read all primary objects in these fields.
+    tic()
+    duplicate_policy = primary_initialization ? :primary : :first
+    catalog = SDSSIO.read_photoobj_files(rcfs, stagedir,
+                        duplicate_policy=duplicate_policy)
+    timing.read_photoobj = toq()
+    Log.info("$(length(catalog)) primary sources")
+
+    reserve_thread[] && thread_fun(reserve_thread)
```

Comment:
Are all these `reserve_thread[] && ...` things necessary? @kpamnany --- Does Dtree need a free thread or something or what's the idea here? Presumably we'd rather use all the threads.